_posts/en/newsletters/2024-03-13-newsletter.md

```diff
@@ -0,0 +1,166 @@
+the meeting.*
+
+[Re enable `OP_CAT`][review club bitcoin-inquisition 39]
+is a PR by Armin Sabouri (GitHub 0xBEEFCAF3) that reintroduces the
+[OP_CAT][topic op_cat] opcode but only on
+[signet][topic signet] [Bitcoin Inquisition][bitcoin inquisition repo]
+and only for [tapscript][topic tapscript] (taproot script).
+Satoshi Nakamoto disabled this opcode in 2010, probably
+out of an overabundance of caution. The operation replaces the
+top two elements on the script evaluation stack with the concatenation
+of those elements.
+
+The motivations for `OP_CAT` were not discussed.
+
+{% include functions/details-list.md
+  q0="What are the various conditions under which the execution of
+      `OP_CAT` may result in failure?"
+  a0="Fewer than 2 items on the stack, resulting item is too large,
+      disallowed by script verify flags (for example, the soft fork is
+      not activated yet), and appears in a non-taproot script (witness
+      version 0 or legacy)."
+  a0link="https://bitcoincore.reviews/bitcoin-inquisition-39#l-46"
+
+  q1="`OP_CAT` redefines one of the `OP_SUCCESSx` opcodes.
+      Why doesn't it redefine one of the `OP_NOPx` opcodes (which has
+      also been used to implement soft fork upgrades in the past)?"
+  a1="Both `OP_SUCCESSx` and `OP_NOPx` opcodes can be redefined
+      to implement soft forks because they restrict the validation
+      rules (they always succeed while the redefined opcodes may fail).
+      Since script execution continues following an `OP_NOP`,
+      redefined `OP_NOP` opcodes can't affect the execution stack
+      (otherwise scripts that used to fail could succeed, which
+      would loosen the rules).
+      Redefined `OP_SUCCESS` opcodes can affect the stack, because
+      `OP_SUCCESS` immediately terminates the script (successfully).
+      Since `OP_CAT` needs to affect the stack, it can't redefine
+      one of the `OP_NOP` opcodes."
+  a1link="https://bitcoincore.reviews/bitcoin-inquisition-39#l-33"
+
+  q2="This PR adds both `SCRIPT_VERIFY_OP_CAT` and
+      `SCRIPT_VERIFY_DISCOURAGE_OP_CAT`. Why are both needed?"
+  a2="It allows the soft fork to be phased in. First both are set to `true`
+      (consensus-enabled but not relayed or mined)
+      until most network nodes are upgraded.
+      Then `SCRIPT_VERIFY_DISCOURAGE_OP_CAT` is set to `false` to enable
+      actual usage. If the Bitcoin Inquisition experiment later fails, the
+      process can be run in reverse.
+      If both are set to `false`, `OP_CAT` is just `OP_SUCCESS`."
+  a2link="https://bitcoincore.reviews/bitcoin-inquisition-39#l-60"
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Core Lightning v24.02.1][] is a minor update to this LN node
+  containing "a few minor fixes [and] improvements in the cost function
+  of the routing algortithm."
```

Comment:
```suggestion
  of the routing algorithm."
```

Reply:
Going to leave this since its a quote from somewhere else

Reply:
Ok nvm, I see Dave already accepted this suggestion